### DISCUSSION TITLE
Thread safety of tycho-document-bundle-plugin

### DISCUSSION THREAD

**Discussion by viltbalint** [eclipse-tycho/tycho#1172] [Q&A]
We are using parallel maven build in our project and got a warning message because the "tycho-document-bundle-plugin" is not marked as thread safe. Is it really a non-thread safe maven plugin? If yes, then do you have any suggestions to keep parallel maven build for the project without disabling this plugin?

**Comment by laeubi** [comment]
@viltbalint we could make/mark it thread save, the usual idiom is to create a lock object and synchronize on the run method of that mojo and then mark it as thread-safe, do you like to provide a PR for this?

**Reply by viltbalint** [reply]
If it is the fastest way to make a releasable version, then I can implement and create a PR for it. Could you provide a reference implementation if possible?

**Reply by laeubi** [reply]
If this is crucial to your business and likes to speed up the development in that area a [sponsoring](https://github.com/sponsors/laeubi?frequency=one-time) would allow me to assign more time-slots to this particular issue. Beside that this project is mostly driven by volunteers, and the best is to checkout the code and start hacking :-)

**Reply by mickaelistria** [reply]
@viltbalint See https://github.com/eclipse/tycho/blob/fee4d3e37df9923583c0e4099a5030dcf18232b0/tycho-p2/tycho-p2-plugin/src/main/java/org/eclipse/tycho/plugins/p2/UpdateSiteP2MetadataMojo.java about an example of how locking can be achieved.
